src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -151,26 +152,36 @@ public IWebHost Build()
             }
             _webHostBuilt = true;
 
-            // Warn about deprecated environment variables
-            if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
-            {
-                Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
-            }
+            var hostingServices = BuildCommonServices(out var hostingStartupErrors);
+            var applicationServices = hostingServices.Clone();
+            var hostingServiceProvider = hostingServices.BuildServiceProvider();
 
-            if (Environment.GetEnvironmentVariable("ASPNET_ENV") != null)
+            if (_options.WriteStatusMessages)
             {
-                Console.WriteLine("The environment variable 'ASPNET_ENV' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
+                // Warn about deprecated environment variables
+                if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
+                {
+                    Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
```

Comment:
Do we intend to ever remove these fallbacks? Also, should we log this instead of writing to the console?

Reply:
We should probably go all or nothing here, either using Console.Writeline or the logger.

Reply:
Let's just remove these, I haven't seen a lot of trouble with then recently.

Reply:
Remove the fallbacks or just remove the messages?

Reply:
Filed https://github.com/aspnet/Hosting/issues/1233 for removing the fallbacks in the next major release.